### PR TITLE
Reject solaris + clang 6.0.0

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

**Comment by gperciva** [comment]
tl;dr: reject clang 6.0.0 on Solaris for all Tarsnap software.

Background: https://github.com/Tarsnap/libcperciva/issues/320

Our software is aimed at C99/POSIX systems.  We can make allowances for some small violations, but there's a whole host of subtle differences between the 30-year-old C90 and the 21-year-old C99.  Trying to enumerate and handle each of those differences would be an exhausting and bug-prone task.

It would also be fairly pointless, as Solaris works with `cc`, `c99`, and `gcc`.  The problematic interaction between clang and Solaris was fixed in clang upstream, so I assume that at some point Solaris will update their `clang` package and this will all be ok.
